### DISCUSSION TITLE
Swing/Amount for jazz?

### DISCUSSION THREAD

**Discussion by ostov-larion** [alda-lang/alda#448] [Ideas]
In the alda documentation I did not find the "swing" attribute. Swing not implemented in Alda?

**Comment by daveyarwood** [comment]
There is no such attribute -- I think it would be a good feature to add :+1: 

Maybe the value could be a number between 0 and 100, representing the percentage of the beat that the first eighth note in a pair  of eighth notes occupies. The default value would be 50, representing equal (50% - 50%) eighth notes.

`(swing 66)` would result in typical swung eighth notes, where a pair of eighth notes is played like 2/3 (66%) of a beat and 1/3 of a beat (33%).

You could also customize the ratio to get different kinds of swing feels. For example, `(swing 60)` would give you "quintuplet swing," a popular sound in recent years where the ratio of the beat for the first eighth note is 3/5 and for the second it's 2/5.

**Comment by daveyarwood** [comment]
This feature would be slightly complicated to implement in that Alda doesn't have time signatures or a notion of where the downbeats are.

I suppose we could assume that wherever the `(swing ...)` attribute is in the score, that point in time is a downbeat, and from there, we can know whether a subsequent note falls on a downbeat by doing some math based on the offset of the initial downbeat and the tempo itinerary.

It also isn't clear to me when we should apply this sort of time stretching. It seems pretty clear in the case of 8th notes that we should stretch out the first eighth note and shrink the second. But what about sixteenth notes? What do jazz musicians do in performance settings? I'm sure somebody has analyzed this before, it's just a matter of finding it.

**Comment by wiwikuan** [comment]
You can use the cram expression to swing: 

For example, `piano: o4 {c4. d4}4 {e4. f4}4 g2` gets you 60%-40% swing 8th notes.
